debezium-core/src/main/java/io/debezium/transforms/UnwrapFromEnvelope.java

```diff
@@ -58,24 +59,61 @@
             .withDescription("Drop delete records converted to tombstones records if a processing connector "
                     + "cannot process them or a compaction is undesirable.");
 
+    private static final Field HANDLE_DELETES = Field.create("delete.handling.mode")
+            .withDisplayName("Handle delete records")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDefault(true)
+            .withDescription("How to handle delete records. Options are: "
+                    + "none - records are passed,"
+                    + "drop - records are removed,"
+                    + "rewrite - __delete field is added to records.");
+
     private boolean dropTombstones;
     private boolean dropDeletes;
-    private final ExtractField<R> delegate = new ExtractField.Value<R>();
+    private String handleDeletes;
+    private final ExtractField<R> afterDelegate = new ExtractField.Value<R>();
+    private final ExtractField<R> beforeDelegate = new ExtractField.Value<R>();
+    private final InsertField<R> removedDelegate = new InsertField.Value<R>();
+    private final InsertField<R> updatedDelegate = new InsertField.Value<R>();
 
     @Override
     public void configure(final Map<String, ?> configs) {
         final Configuration config = Configuration.from(configs);
-        final Field.Set configFields = Field.setOf(DROP_TOMBSTONES, DROP_DELETES);
+        final Field.Set configFields = Field.setOf(DROP_TOMBSTONES, HANDLE_DELETES);
         if (!config.validateAndRecord(configFields, logger::error)) {
             throw new ConnectException("Unable to validate config.");
         }
 
         dropTombstones = config.getBoolean(DROP_TOMBSTONES);
         dropDeletes = config.getBoolean(DROP_DELETES);
```

Comment:
`DROP_DELETES` should no longer have default value and when present a warning should be logged.

Reply:
That could a problem, because it's boolean so we have only 2 values: true, false.
And we can't tell when false is false or unset.

Reply:
Would not `config.hasKey()` solve this problem?

Reply:
You're right.